Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,403 @@
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<Calendar>> GetCalendarsAsync()
+		{
+			return Task.Run<IList<Calendar>>(() =>
+			{
+				CalendarRecord calRecord = new CalendarRecord(Book.Uri);
+				var calendars = new List<Calendar>();
+
+				calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+				calendars.Add(
+					new Calendar
+					{
+						Name = calRecord.Get<string>(Book.Name),
+						ExternalID = calRecord.Get<int>(Book.Id).ToString(),
+						CanEditEvents = true,
+						CanEditCalendar = false,
+						AccountName = calRecord.Get<int>(Book.AccountId).ToString(),
+						Color = calRecord.Get<string>(Book.Color),
+					}
+				);
+
+				calRecord?.Dispose();
+				calRecord = null;
+
+				return calendars;
+			});			
+		}
+
+		/// <summary>
+		/// Gets a single calendar by platform-specific ID.
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar identifier</param>
+		/// <returns>The corresponding calendar, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<Calendar> GetCalendarByIdAsync(string externalId)
+		{
+			CalendarRecord calRecord = new CalendarRecord(Book.Uri);
+			Calendar calendar = new Calendar();
+
+			calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+			calendar.Name = calRecord.Get<string>(Book.Name);
+			calendar.ExternalID = calRecord.Get<int>(Book.Id).ToString();
+			calendar.CanEditEvents = true;
+			calendar.CanEditCalendar = false;
+			calendar.AccountName = calRecord.Get<int>(Book.AccountId).ToString();
+			calendar.Color = calRecord.Get<string>(Book.Color);
+
+			calRecord?.Dispose();
+			calRecord = null;
+			
+			return Task.Run(() => calendar);
+		}
+
+		/// <summary>
+		/// Gets all events for a calendar within the specified time range.
+		/// </summary>
+		/// <param name="calendar">Calendar containing events</param>
+		/// <param name="start">Start of event range</param>
+		/// <param name="end">End of event range</param>
+		/// <returns>Calendar events</returns>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
+		{
+			return Task.Run<IList<CalendarEvent>>(() =>
+			{
+				var calendars = new List<CalendarEvent>();
+
+				CalendarManager calManager = new CalendarManager();
+				CalendarQuery calQuery = new CalendarQuery(Event.Uri);
+				CalendarList calList = calManager.Database.GetRecordsWithQuery(calQuery, 0, 0);
```

Comment:
The point of my suggesting the use of `GetRecordsWithQuery` was to actually pass in a `CalendarFilter` that filters on start/end times to the query, not to just use an empty query.

Such as in the example from the docs that uses `filter.AddCondition(CalendarFilter.LogicalOperator.And, CalendarViews.Event.Start, CalendarFilter.IntegerMatchType.GreaterThanOrEqual, start);`